nexus/src/app/oximeter.rs

```diff
@@ -212,14 +216,100 @@ impl super::Nexus {
             .map_err(|e| Error::internal_error(&e.to_string()))?
             .timeseries_schema_list(&pag_params.page, limit)
             .await
-            .map_err(|e| match e {
-                oximeter_db::Error::DatabaseUnavailable(_) => {
-                    Error::ServiceUnavailable {
-                        internal_message: e.to_string(),
-                    }
+            .map_err(map_oximeter_err)
+    }
+
+    /// Returns a results from the timeseries DB based on the provided query
+    /// parameters.
+    ///
+    /// * `timeseries_name`: The "target:metric" name identifying the metric to
+    /// be queried.
+    /// * `criteria`: Any additional parameters to help narrow down the query
+    /// selection further.
+    /// * `query_params`: Pagination parameter, identifying which page of
+    /// results to return.
+    /// * `limit`: The maximum number of results to return in a paginated
+    /// request.
+    pub async fn select_timeseries(
+        &self,
+        timeseries_name: &str,
+        criteria: &[&str],
+        query_params: ResourceMetricsPagination,
```

Comment:
To make sure I understand, this type is used to express the start/end range of the timeseries samples, as a time range, correct? E.g., `/other/url/stuff?start_time=SOME_TIME?end_time=SOME_LATER_TIME`. Does that mean both need to be specified? The querying interface in `oximeter_db` itself allows either or both to be optional, though that's not to say we _need_ to support optional time bounds.

Reply:
I made it non-optional because I'm not aware of a use case where you'd need an unbounded timeframe, and it's easier to make it optional later than it is to make it optional now and change that later.

Reply:
Fair enough! I think leaving it as-is works for now, I was mostly curious.

One possible situation (maybe?) where we'd want an unbounded time interval is when the consumer doesn't _know_ the start time. E.g., if we want to start looking at disk writes from whenever the disk was first created, we could do that by either looking up the creation time, or just not specifying the start time. Or I guess using the Unix epoch 😆